### PR TITLE
add osx_arm64

### PR DESCRIPTION
**Pull request by garymm**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by garymm** [comment]
@conda-forge-admin, please rerender

**Comment by garymm** [comment]
Hmm seems something in the build tries to execute a build output and fails b/c of arch mismatch... May take a look later.

**Comment by garymm** [comment]
OK I'm pretty stumped. The issue is the build tries to execute the capnp tool.
Executing it from the output of the build itself doesn't work because we're cross-compiling.
Trying to get mamba to install a compatible version ahead of time so we can use that doesn't seem to work because of version conflicts with other packages.
Current version is trying to use mamba during the build script, but it seems to have hung for some reason.
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=865053&view=logs&j=ced5d8de-8227-5f3f-33a1-45cf1592c45a

Any suggestions?

**Comment by garymm** [comment]
Aha it's working!

File: recipe/meta.yaml
Comment (LehMaxence) [inline recipe/meta.yaml:1]: Could you bump the build number to `1` on line 13? Then it should be ready to merge.

File: recipe/build.sh [hunk 1]
```diff
@@ -9,9 +9,25 @@ if [[ $target_platform == "linux-"* ]]; then
...
 
-make -j${CPU_COUNT} check
-make install
\ No newline at end of file
+configure_cmd=(./configure --enable-shared --prefix=$PREFIX)
+
+# cross-compiling needs to use prebuilt capnp
+# https://github.com/capnproto/capnproto/issues/1815#issuecomment-1732327995
+if [[ "${BUILD_ARCH}" != "${HOST_ARCH}" ]]; then
+    mamba install --no-deps --yes capnproto
```
Comment (isuruf) [inline recipe/build.sh:24]: You should add this to `build: - capnproto # [build_platform != target_paltform]`

Comment (garymm) [reply recipe/build.sh:24]: I tried but installation fails because of openssl incompatibility with other required packages

Comment (garymm) [reply recipe/build.sh:24]: Maybe would be fixed by https://github.com/conda-forge/capnproto-feedstock/pull/32

**Comment by martin-g** [comment]
@conda-forge-admin, please ping team

**Pull request merged**